nemo_skills/dataset/ojbench/prepare.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import shutil
+import subprocess
+import sys
+from pathlib import Path
+
+REPO_URL = "https://huggingface.co/datasets/He-Ren/OJBench_testdata"
+HF_TOKEN = os.environ.get("HF_TOKEN")
+
+
+def clone_dataset_repo(url, destination):
+    if not shutil.which("git"):
+        print("❌ Error: Git executable not found. Please install Git.", file=sys.stderr)
+        sys.exit(1)
+
+    try:
+        if destination.exists() or destination.is_symlink():
+            print(f"Destination '{destination}' already exists. Removing it...")
+            if destination.is_dir():
+                shutil.rmtree(destination)
+            else:
+                destination.unlink()
+
+        auth_url = url.replace("https://huggingface.co/", f"https://user:{HF_TOKEN}@huggingface.co/", 1)
```

Comment:
why do we need to clone it? Can't we use datasets api instead and save the data through that? The rabbit's comment about HF_TOKEN leakage is actually an important thing to fix and should be done automatically if we don't add it to repo url

Reply:
@Kipok I think we can't use the datasets api because the repo is not configured to be used in that way.